The RC4 stream cipher is required but openssl3 does not include it in the
default provider. Since the legacy provider includes RC4, we can load it
here, alongside the default.
--- a/libtransmission/crypto-utils-openssl.c
+++ b/libtransmission/crypto-utils-openssl.c
@@ -20,6 +20,7 @@
 #include <openssl/rand.h>
 #include <openssl/ssl.h>
 #include <openssl/x509.h>
+#include <openssl/provider.h>

 #include "transmission.h"
 #include "crypto-utils.h"
@@ -184,6 +185,8 @@

 tr_rc4_ctx_t tr_rc4_new(void)
 {
+    OSSL_PROVIDER_load(NULL, "legacy");
+    OSSL_PROVIDER_load(NULL, "default");
     EVP_CIPHER_CTX* handle = EVP_CIPHER_CTX_new();

     if (check_result(EVP_CipherInit_ex(handle, EVP_rc4(), NULL, NULL, NULL, -1)))
